src/index_emails.py

```diff
@@ -82,24 +84,74 @@ def create_index():
 def upload_batch(upload_data):
     upload_data_txt = ""
     for item in upload_data:
-        cmd = {'index': {'_index': tornado.options.options.index_name, '_type': 'email', '_id': item['message-id']}}
+        cmd = {'index': {'_index': tornado.options.options.index_name, '_type': tornado.options.options.es_document_type, '_id': item['message-id']}}
         upload_data_txt += json.dumps(cmd) + "\n"
         upload_data_txt += json.dumps(item) + "\n"
 
-    request = HTTPRequest(tornado.options.options.es_url + "/_bulk", method="POST", body=upload_data_txt, request_timeout=240)
-    response = http_client.fetch(request)
-    result = json.loads(response.body)
+    upload_successful = False
+    upload_attempts = 0
+    res_txt = ""
+    while ((not upload_successful) and (upload_attempts < tornado.options.options.es_upload_attempts)):
 
-    global total_uploaded
-    total_uploaded += len(upload_data)
-    res_txt = "OK" if not result['errors'] else "FAILED"
-    logging.info("Upload: %s - upload took: %4dms, total messages uploaded: %6d" % (res_txt, result['took'], total_uploaded))
+        upload_attempts = upload_attempts + 1
+
+        try:
+            request = HTTPRequest(tornado.options.options.es_url + "/_bulk", method="POST", body=upload_data_txt, request_timeout=tornado.options.options.es_http_timeout_seconds)
+            response = http_client.fetch(request)
+            result = json.loads(response.body)
+
+            if result['errors']:
+                logging.error("upload attempt: %d FAILED: %s" % (upload_attempts,response.body))
+
+            else:
+                upload_successful = True
+                res_txt = "OK"
+        except:
+            logging.error("upload attempt: %d FAILED: %s %s" % (upload_attempts,sys.exc_info()[0],sys.exc_info()[1]))
 
 
+    if upload_successful:
+        global total_uploaded
+        total_uploaded += len(upload_data)
+        logging.info("Upload: %s - upload took: %4dms, total messages uploaded: %6d" % (res_txt, result['took'], total_uploaded))
+    else:
+        logging.error("All upload attempts failed, see previous error messages")
+
 def normalize_email(email_in):
     parsed = email.utils.parseaddr(email_in)
     return parsed[1]
 
+def content_type_is_ignored(content_type):
+    if tornado.options.options.index_bodies_ignore_content_types:
+        ignore_types = tornado.options.options.index_bodies_ignore_content_types.split(",")
+        for to_ignore in ignore_types:
+            if to_ignore:
+                if to_ignore.strip().lower() in content_type.lower():
```

Comment:
Should move the `lower()` out of the loop so it's not called for each call to `content_type_is_ignored()`.

Reply:
done